File: frontend/lib/models/send_message_request.dart
class SendMessageRequest {
  final String senderId;
  final String receiverId;
  final String content;
  final String type;

  SendMessageRequest({
    required this.senderId,
    required this.receiverId,
    required this.content,
    required this.type,
  });

  Map<String, dynamic> toJson() => {
    'senderId': senderId,
    'receiverId': receiverId,
    'content': content,
    'type': type,
  };

  factory SendMessageRequest.fromJson(Map<String, dynamic> json) {
    return SendMessageRequest(
      senderId: json['senderId'],
      receiverId: json['receiverId'],
      content: json['content'],
      type: json['type'],
    );
  }
}

File: backend/src/main/java/com/example/chatify/message/SendMessageRequest.java
package com.example.chatify.message;

import com.example.chatify.enums.MessageType;

import java.util.UUID;

public class SendMessageRequest {
    private UUID senderId;
    private UUID receiverId;
    private String content; // Can be text or a file URL
    private MessageType type;

    public UUID getSenderId() {
        return senderId;
    }

    public void setSenderId(UUID senderId) {
        this.senderId = senderId;
    }

    public UUID getReceiverId() {
        return receiverId;
    }

    public void setReceiverId(UUID receiverId) {
        this.receiverId = receiverId;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public MessageType getType() {
        return type;
    }

    public void setType(MessageType type) {
        this.type = type;
    }
}
